numba/types/containers.py

```diff
@@ -300,6 +300,23 @@ def unify(self, typingctx, other):
                 return Tuple(unified)
 
 
+class StarArgTuple(Tuple):
+    """To distinguish from Tuple() used as argument to a `*args`.
+    """
+    def __new__(cls, types):
+        _HeterogeneousTuple.is_types_iterable(types)
+
+        if types and all(t == types[0] for t in types[1:]):
+            return StarArgUniTuple(dtype=types[0], count=len(types))
+        else:
+            return object.__new__(StarArgTuple)
+
+
+class StarArgUniTuple(UniTuple):
+    """To distinguish from UniTuple() used as argument to a `*args`.
+    """
```

Comment:
Should these get a custom `name` written to `self.name` so as to distinguish the type in traceback/debug?

Reply:
good catch. yes it needs new name

Reply:
fixed in https://github.com/numba/numba/pull/4978/commits/90956ff11c979177e617d38e42a2bae56a4ac54a